inversify.config.ts

```diff
@@ -0,0 +1,144 @@
+
+// Message Management
+import MessageHandlerFactory from './source/MessageHandlerFactory';
+import IMessageHandler from './source/IMessageHandler';
+import SendDataHandler from './source/MessageHandlers/SendDataHandler';
+import PreviewHandler from './source/MessageHandlers/PreviewHandler';
+
+// Messages
+import IMessage from './source/IMessage';
+import MessageFactory from './source/MessageFactory';
+import SendDataMessage from './source/Messages/SendDataMessage';
+import PreviewMessage from './source/Messages/PreviewMessage';
+
+// Data Access
+import IDataReader from './source/DataAccess/IDataReader';
+import IDataWriter from './source/DataAccess/IDataWriter';
+import S3Writer from './source/DataAccess/S3/S3Writer';
+import OracleReader from './source/DataAccess/Oracle/OracleReader';
+import ICommand from './source/Commands/ICommand';
+import AdHocIngestCommand from './source/Commands/AdHocIngestCommand';
+import AdHocPreviewCommand from './source/Commands/AdHocPreviewCommand';
+
+// TODO: Make configurable?
+const REGION = 'us-east-1';
+const stage = 'prod';
+
+const container = new Container();
+
+// TODO: Get region
+const s3Config: S3.ClientConfiguration = { region: REGION };
+const sqsConfig: SQS.ClientConfiguration = { apiVersion: '2012-11-05', region: REGION };
+
+const logger: Winston.Logger = Winston.createLogger({
+    format: Winston.format.json(),
+    level: 'info',
+    transports: [
+        new Winston.transports.Console()
+    ]
+});
+process.env.DISCOVERY_SERVICE = 'https://4w35qhpotd.execute-api.us-east-1.amazonaws.com/prod';
+const sdk: DiscoverySdk = new DiscoverySdk(process.env.DISCOVERY_SERVICE, REGION);
+
+const poolLocator = new CognitoUserPoolLocatorUserManagement(REGION);
+const authManager = new AuthManager(poolLocator, REGION);
+let queueUrl: string;
+let tenantId: string;
+
+const s3Buckets = {
+    dev: 'adastra-dev-data-ingestion',
+    prod: 'adastra-prod-data-ingestion'
+};
+
+const bucketName = s3Buckets[stage];
+
+// NOTE: updates to the discovery service itself would require pushing a new docker image.
+// This should still be an environment variable rather than hardcoded
+process.env.DISCOVERY_SERVICE = 'https://4w35qhpotd.execute-api.us-east-1.amazonaws.com/prod';
+
+const startup =
+    () => (async () => {
+        const endpoints = await sdk.lookupService('user-management', stage);
+        process.env.USER_MANAGEMENT_URI = endpoints[0];
+    })()
+    .then(async () => {
+        const cognitoJwt = await authManager.signIn(process.env.ASTRA_CLOUD_USERNAME, process.env.ASTRA_CLOUD_PASSWORD);
+
+        // Get IAM credentials
+        const iamCredentials = await authManager.getIamCredentials(cognitoJwt.idToken);
+
+        // Set up authenticated access to SQS
+        sqsConfig.credentials = {
+            accessKeyId: iamCredentials.AccessKeyId,
+            secretAccessKey: iamCredentials.SecretKey,
+            sessionToken: iamCredentials.SessionToken
+        };
```

Comment:
These creds don't belong in a global static initialization application lifecycle, but I can address that in a separate PR